src/request/import/ImportApi.js

```diff
@@ -18,12 +19,16 @@ class ImportApi extends TopLevelApi {
         parsedRequest.requestedKeyPaths = this.parsePathsArray(request.requestedKeyPaths, ' requestedKeyPaths');
         parsedRequest.isKeyLost = this.parseBoolean(request.isKeyLost);
         parsedRequest.enableBackArrow = this.parseBoolean(request.enableBackArrow);
+        parsedRequest.wordsOnly = this.parseBoolean(request.wordsOnly);
+        parsedRequest.expectedKeyId = request.expectedKeyId ? await this.parseKeyId(request.expectedKeyId) : undefined;
```

Comment:
As we're actually using only the key id and not the whole `KeyInfo` and the property is called `expectedKeyId`, I would assign only the `.id` here.

Also: `parseKeyId` throws for an unknown key id which does not necessarily make sense in the import case.
I would suggest:

```suggestion
        parsedRequest.expectedKeyId = request.expectedKeyId ? request.expectedKeyId.toString() : undefined;
```

Reply:
This is the same as an remove-key followed by an import, except for this very `expectedKeyId`. The `expectedKeyId` is only ever used for the specific purpose of resetting your password of an existing Account. 
If that account does not exist, or the keyId is something like `'-'` other tests in this request will always fail, as in this case you are only allowed to import with the recovery words which result in this specific id. 
The `expectedKeyId` is of course also expected to exist anyways, as it existed a few moments ago, but I think the parsing should not rely on that.

I could change it to 
```suggestion
        parsedRequest.expectedKeyId = request.expectedKeyId ? (await this.parseKeyId(request.expectedKeyId)).id : undefined;
```
and update the type accordingly, but I would keep the parsing as it is, because I think the throw is the desired behaviour here.

Reply:
By "unknown key id" I meant a valid key id string for which no key exists in the database.
For this case, `parseKeyId` throws a `KeyNotFoundError`.
However, I think that for an import request it's totally acceptable that the key that I want to import is not currently present in the database. All we care about is that the `expectedKeyId` matches for the key resulting from the words the user entered.
I think the change I suggested serves this purpose.
If I'm not mistaken (correct me if I'm wrong) the following holds true: `(await this.parseKeyId(someId)).id === someId` for a Key existing in the new `KeyStore`.
For keys in the `AccountStore` this does not hold true which should however not be of relevance as such do not exist anymore after the migration process.
If this is the case, we should be able to replace the term `(await this.parseKeyId(request.expectedKeyId)).id` with `request.expectedKeyId.toString()` to remove the dependency on the existence of the key in the database.

Reply:
In the case the `expectedKeyId` is set this is basically a pseudo change-password. It is not really an import, but a re-import of an existing account. I do not see a benefit in opening it up to arbitrary keyIds or arbitrary strings for that matter. They could end up giving the user a usable interface, which he is unable to complete. If this is used as it is intended this will most likely never throw, as the request comes from a change-password to this specific keyId moments ago, which in turn did not throw for the same validation. If it is used as it was not intended it is going to throw.

What use-case do you have in mind for a non-existing keyId?

Reply:
Now resolved by separate `ResetPasswordRequest` type.